app/services/orchestration/task.rb

```
@@ -1,12 +1,17 @@
 module Orchestration
   class Task
-    attr_reader :name, :status, :priority, :action, :timestamp
+    attr_reader :id, :name, :status, :priority, :action, :timestamp, :created
 
     def initialize(opts)
       @name      = opts[:name]
+      @id        = opts[:id].try(:to_s) || @name
       @status    = opts[:status]
       @priority  = opts[:priority] || 0
       @action    = opts[:action]
+      @created   = opts[:created] || Time.now.to_f
+      raise("action must be present for task '#{name}'") if action.nil?
+      # this kills tests, need to update other tests before
+      #Foreman::Deprecation.deprecation_warning('1.19', "Orchestration task '#{name}' is missing required :id field") if @id.nil?
```

should this still be commented?

---

I removed those, my initial idea was to fail fast with deprecation warning to enforce having `id` but I think nobody will notice this in test output, it's better I think to have sane default (use name if id is not set) and go with that.